### PR TITLE
Add confirmation for quitting the game

### PR DESCRIPTION
**Pull request by impaktor**
Add confirmation for quitting the game, in 17 flavours.

A little homage to great game devs of the 90s like ID Software and Apogee (Doom 1-2, Spear of Destiny, Wolfenstein 3D), by having random confirmation questions. It can be disabled, which is perhaps useful when testing and developing.

2015-03-30-210141_859x528_scrot

This probably isn't the right place to put this option, please advice:

2015-03-30-210229_377x128_scrot

### CODE REVIEW

**Comment by fluffyfreak** [comment]
Yes, I want this in the game!

**Comment by robn** [comment]
Cute :shipit:

If the option is only useful for testing and dev, then make it a config setting only. Otherwise, don't worry about the placement. The settings need a bit of information-architecture brains thrown at it, until then chucking everything in one basket is probably fine.

File: data/lang/quitconfirmation-core/en.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      "description" : "No, don't quit program",
+      "message" : "No"
+   },
+   "MSG_1" : {
+      "description" : "",
+      "message" : "Get out of here and go back to your boring programs?"
+   },
+   "MSG_2" : {
+      "description" : "",
+      "message" : "Don't go now, there's a Pirate waiting for you in the other program."
```
Comment (robn) [inline data/lang/quitconfirmation-core/en.json]: Lowercase P on pirate.

File: data/lang/quitconfirmation-core/en.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      "description" : "",
+      "message" : "Are you sure you want to quit this great game that cost you nothing?"
+   },
+   "MSG_4" : {
+      "description" : "",
+      "message" : "Doest thou wish to leave with such hasty abandon?"
+   },
+   "MSG_5" : {
+      "description" : "(Inspired by Wolfenstein 3D)",
+      "message" : "Choose {no} for guns and glory, {yes} for work and worry."
```
Comment (robn) [inline data/lang/quitconfirmation-core/en.json]: NO and YES better?

File: data/lang/quitconfirmation-core/en.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      "description" : "",
+      "message" : "Please choose {no} to waste a few more productive hours. Please?"
+   },
+   "MSG_11" : {
+      "description" : "",
+      "message" : "Explorers don't quit, but go ahead and click {yes} if you aren't one."
+   },
+   "MSG_12" : {
+      "description" : "",
+      "message" : "Choose {no} If you are a though bounty hunter, if not, click {yes} daintily."
```
Comment (robn) [inline data/lang/quitconfirmation-core/en.json]: thorough?

**Comment by robn** [comment]
Doesn't work for if you quit by closing the window, of course. I think that's fine - its quite awkward to do. Just saying :)

File: data/lang/quitconfirmation-core/en.json (hunk 1 above)
Comment (impaktor) [inline data/lang/quitconfirmation-core/en.json]: Fixed.
Thanks.
Will push later.

File: data/lang/quitconfirmation-core/en.json (hunk 2 above)
Comment (impaktor) [inline data/lang/quitconfirmation-core/en.json]: Ironically, I'm thinking both yes and no.

Uppercase does have its merits, but in other missions we have `{price}`, not `{PRICE}`, so just trying to be consistent.

If that was what you meant?

File: data/lang/quitconfirmation-core/en.json (hunk 3 above)
Comment (impaktor) [inline data/lang/quitconfirmation-core/en.json]: Doh! should be "Tough". 
Thanks. 
Will fix.
(Hacked it all up today in "the between time" at work.)

File: data/lang/quitconfirmation-core/en.json (hunk 2 above)
Comment (robn) [inline data/lang/quitconfirmation-core/en.json]: Oh, these are interpolated values. I withdraw my stupid comment.

**Comment by impaktor** [comment]
Updated and pushed the fixes.

> If the option is only useful for testing and dev, then make it a config setting only.

I was just speculating as to who/why people might object to this, so I made it optional. I'll leave it in for now, as you say, it is easy to move or change it later.

I do point out that this continues the (new) tradition of not pouring everything into `ui-core` or `core`, but put it in a new `quitconfirmation-core` module. Perhaps not the best name? Perhaps have `ui-` somewhere in the name as well?

Anyway I'll add the module to transifex when merging.

**Comment by walterar** [comment]
I like this :+1:

**Pull request merged**